### PR TITLE
{bio}[foss/2016b] IQ-TREE 1.5.5

### PR DESCRIPTION
**Pull request by ysagon**

### CODE REVIEW

**Comment by ysagon** [comment]
@boegel please review!

File: easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+easyblock = 'CMakeMake'
+name = 'IQ-TREE'
```
Comment (boegel) [inline easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb]: style nitpicking: please include an empty line above `name =`

File: easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+homepage = 'http://www.iqtree.org/'
+description = """Efficient phylogenomic software by maximum likelihood"""
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+github_account = 'Cibiv'
+
+source_urls = [GITHUB_SOURCE]
+sources = ['v%(version)s.tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb]: please include a SHA256 checksum (pro tip: use `eb --inject-checksums`)

File: easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+configopts = '-DIQTREE_FLAGS="mpi omp" -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx'
+
+builddependencies = [('CMake', '3.5.2')]
+
+sanity_check_paths = {
+    'files': ["bin/iqtree-omp-mpi"],
+    'dirs': ["bin"],
```
Comment (boegel) [inline easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb]: no need to check for a non-empty `bin` if you're already checking for a binary file in that subdirectory

File: easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+description = """Efficient phylogenomic software by maximum likelihood"""
+
+toolchain = {'name': 'foss', 'version': '2016a'}
+
+github_account = 'Cibiv'
+
+source_urls = [GITHUB_SOURCE]
+sources = ['v%(version)s.tar.gz']
+
+configopts = '-DIQTREE_FLAGS="mpi omp" -DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx'
```
Comment (boegel) [inline easybuild/easyconfigs/i/IQ-TREE/IQ-TREE-1.5.5-foss-2016a-mpi.eb]: don't hardcode these compiler (wrapper) commands, enable the `usempi` toolchain option instead:

```python
toolchain = {'name': 'foss', 'version': '2016a'}
toolchainopts = {'usempi': True}

**Comment by boegel** [comment]
@ysagon You may want to retain `configopts = "-DIQTREE_FLAGS="mpi omp"`?

Have you tested this easyconfig as is?

**Comment by ysagon** [comment]
It seems that without the configopts, it's the mpi only version that is build.
I have thus specified omp and mpi, and renamed the eb file accordingly. For me the build produce a correct binary.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2075.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4068f93719752db5ec21c02eae39d29d for a full test report.

**Comment by ysagon** [comment]
Can you help me to understand the error? It seems all the test made by travis are ok but one is not (your test report).
And I have no clue why it's not passing the build.

Maybe it's for an optimization reason? I'm compiling all my eb file using 

    optarch = march=core2

Maybe there is a problem if we don't force the compilation to use this arch?

**Comment by boegel** [comment]
@ysagon I'll give it a shot with `eb --optarch=march=core2 ...`...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2075.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/c5eb56b7f51984df3b0c8a02c7b61761 for a full test report.

**Comment by boegel** [comment]
@ysagon That seems to help indeed, let's try disabling architecture-specific optimisations:

```python
# architecture-specific optimisations must be disabled in order for build to work
toolchainopts = {'usempi': True, 'optarch': False}
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2075.delcatty.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/3971bc1541573db44d7cd83d92fb53fd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2514.golett.os - Linux centos linux 7.4.1708, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/a6a006b1b2429e87d3f6ea820148d685 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @ysagon!

**Pull request merged**